src/python/pants/backend/jvm/subsystems/jvm_platform.py

```diff
@@ -149,6 +165,33 @@ def get_platform_for_target(self, target):
         return platform
     return self.get_platform_by_name(target.payload.platform, target)
 
+  def get_runtime_platform_for_target(self, target):
+    """Find the runtime platform associated with this target.
+
+    :param JvmTarget target: target to query.
+    :return: The jvm platform object.
+    :rtype: JvmPlatformSettings
+    """
+    # Lookup order
+    # - target's declared runtime_platform
+    # - default runtime_platform
```

Comment:
Interesting. It's a bit funky to think about having one of these set but not the other, but I guess that it allows for a basic configuration where the "default_platform" is the only one you set, because your usecase doesn't need to differentiate between the two?

The target argument docs are great, but it would be good to expand the help for the two `--default.*platform` args to explain their relationship a bit better as well.